module/rapiddisk.c

```diff
@@ -844,12 +853,18 @@ static int attach_device(unsigned long num, unsigned long long size)
 	disk->queue->nr_requests = nr_requests;
 	disk->queue->limits.discard_granularity = PAGE_SIZE;
 	disk->queue->limits.max_discard_sectors = UINT_MAX;
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,19,0) || (defined(RHEL_MAJOR) && RHEL_MAJOR >= 9 && RHEL_MINOR >= 0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,11,0)
+	blk_queue_disable_discard(disk->queue);
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(5,19,0) || (defined(RHEL_MAJOR) && RHEL_MAJOR >= 9 && RHEL_MINOR >= 0)
 	blk_queue_max_discard_sectors(disk->queue, 0);
 #else
 	blk_queue_flag_set(QUEUE_FLAG_DISCARD, disk->queue);
 #endif
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,11,0)
+	disk->queue->limits.features = BLK_FEAT_ROTATIONAL;
```

Comment:
Why rotational? Is this the right feature to define? For instance, I am looking at the brd code and it only defines BLK_FEAT_SYNCHRONOUS and BLK_FEAT_NOWAIT. Thoughts?

Reply:
@piso77: I've forwarded your rapiddisk patch from the Ubuntu package to upstream. Upstream has a question that you hopefully can answer.

Reply:
Any updates on this? I'd prefer not to represent the RAM drive as a rotational drive and if anything, I'd rather align with how brd presents the drive.

Reply:
You are right, inventing `BLK_FEAT_ROTATIONAL`  sounds wrong, and seems to be the opposite to setting `QUEUE_FLAG_NONROT` in the `#else` branch. If at all, it should be inherited from the underlying device.

Looking at https://github.com/torvalds/linux/commit/bd4a633b6f7c3c6b6ebc1a07317643270e751a94 I wouldn't have set that flag and only removed the NONROT part. I'll change it that way, but I'll leave setting any  new `BLK_FEAT_*` flags to you.